qualang_tools/bakery/readme.md

```diff
@@ -0,0 +1,273 @@
+
+The logic behind the baking context manager is to stay as close as possible to the way we would write play statements within a QUA program. For instance, commands like frame_rotation, reset_phase, ramp, wait and align are all replicated  within the context manager. 
+
+For playing custom operations, from an arbitrary shaped waveform the procedure goes as follows:
+
+1. You first have to write down the sample you want to use as a waveform (with arbitrary length, no matter if it does not match usual QUA criteria for saving a waveform in memory) in the form of a Python list.
+    - If the sample is meant for a singleInput element, the list should contain the sample itself. 
+    - Contrariwise, if it is intended for a mixInputs element, the list should contain two Python lists as ```[sample_I, sample_Q]``` , where sample_I and sample_Q are themselves Python lists containing the samples.
+
+2. Add the sample to the local configuration, with method ```add_Op```, which takes 4 inputs: 
+    - The name of the operation (name you will use only within the baking context manager in a play statement)
+    - The quantum element for which you want to add the operation
+    - The samples to store as waveforms
+    - The digital_marker name (supposedly already existing in the configuration) to attach to the pulse associated to the operation.
+
+3. Use a baking ``play()`` statement, specifying the operation name (which should correspond to the name introduced in the add_Op method) and the quantum element to play the operation on
+
+All those commands concatenated altogether eventually build one single “big” waveform per quantum element involved in the baking that contains all the instructions specified in the baking environment. The exiting procedure of the baking ensures that the appropriate padding is done to ensure that QUA will be able to play this arbitrary waveform.
+
+Here is a basic code example that simply plays two pulses of short lengths: 
+
+
+```
+with baking(config, padding_method = "symmetric_r") as b:
+
+# Create arbitrary waveforms 
+
+  singleInput_sample = [1., 0.8, 0.6, 0.8, 0.9]
+  mixInput_sample_I = [1.2, 0.3, 0.5]
+  mixInput_sample_Q = [0.8, 0.2, 0.4]
+  
+  # Assign waveforms to quantum element operation
+  
+  b.add_Op("single_Input_Op", "qe1", singleInput_sample, digital_marker= None)
+  b.add_Op("mix_Input_Op", "qe2", [mixInput_sample_I, mixInput_sample_Q], digital_marker = None)
+  
+  # Play the operations
+  
+  b.play("single_Input_Op", "qe1")
+  b.play("mix_Input_Op", "qe2")
+```
+# **How to play in QUA my baked waveforms?**
+
+The baking object has a method called run, which takes no inputs and simply does appropriate alignment between quantum elements involved in the baking and play simultaneously (using this time a QUA play statement) the previously baked waveforms. Therefore, what is left to do is to **call the run method associated to the baking object within the actual QUA program**.

```

Comment:
Therefore, **all that** is left to do is to **call the run method associated to the baking object within the actual QUA program**.